meta-resin-common/recipes-support/balena-custom-udev-rules/balena-custom-udev-rules/balena-custom-udev-rules

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+. /usr/sbin/resin-vars
+
+keys=`jq -r 'select(.settings != null) | select(.settings.udevrules != null) | .settings.udevrules | keys[] ' $CONFIG_PATH`
+RELOAD_RULES=0
+lscmd="ls "
+
+if [ ! -z "$keys" ]; then
+	echo "Found custom udev rules in $CONFIG_PATH"
+	while read -r key; do
+		echo "Found custom udev rule: $key"
+		lscmd="$lscmd -I \"$key.rules\""
+		filter="'.settings.udevrules.\"$key\" []'"
+		cmd="jq -r $filter $CONFIG_PATH"
+		rule=$(eval "$cmd")
+		if [[ -f "/etc/udev/rules.d/$key.rules" ]] && [[ $(< "/etc/udev/rules.d/$key.rules") = "$rule" ]]; then
+			echo "/etc/udev/rules.d/$key.rules already exists"
+			continue
+		else
+			RELOAD_RULES=1
+			echo "New rule or modified rule detected. Writing /etc/udev/rules.d/$key.rules and reloading"
+			echo "$rule" > "/etc/udev/rules.d/$key.rules"
+		fi
+	done <<< "$keys"
+
+        if [ ! -f /etc/udev/rules.d/README ]; then
+                echo "Files here are auto-generated by balena-custom-udev-rules. Extra files will be removed" >> /etc/udev/rules.d/README
+        fi
+
+	lscmd="$lscmd /etc/udev/rules.d"
+	extra_rules=$(eval "$lscmd")
+	if [ "$extra_rules" != "README" ]; then
```

Comment:
This looks redundant as you check later in a per rule loop.

Reply:
removed